File: app/src/main/java/com/example/pam_app/view/BucketView.java
package com.example.pam_app.view;

import com.example.pam_app.model.Bucket;
import com.example.pam_app.utils.listener.Clickable;

public interface BucketView {
    void bind(Bucket bucket);
    void back();
    void goToAddEntry(String bucketName, int bucketType);
    void showGetBucketError();
    void showDeleteBucketError();
    void showDeleteBucketSuccess();
    void showSureDialog(Clickable clickable);
}

File: app/src/main/java/com/example/pam_app/view/BucketView.kt
package com.example.pam_app.view

import com.example.pam_app.model.Bucket
import java.io.Serializable
import java.util.*

interface BucketView {
    fun bind(bucket: Bucket?)
    fun goToAddEntry(bucketName: String?, bucketType: Int)
    fun back(entries: Serializable?, id: Int?)
    fun showGetBucketError()
    fun showDeleteBucketError()
    fun showDeleteBucketSuccess()
    fun showSureDialog(clickable: () -> Unit)
    fun goToAddBucket(bucketId: Int)
}